.gitignore

```diff
@@ -1,47 +1,25 @@
-## generic files to ignore
-*.classpath
-*.project
-*.settings
-target/*
-jt-affine/target/*
-jt-algebra/target/*
-jt-scale/target/*
-jt-warp/target/*
-jt-iterators/target/*
-jt-utilities/target/*
-jt-mosaic/target/*
-jt-translate/target/*
-jt-lookup/target/*
-jt-stats/target/*
-jt-zonal/target/*
-jt-crop/target/*
-jt-rescale/target/*
-jt-nullop/target/*
-jt-bandmerge/target/*
-jt-bandselect/target/*
-jt-border/target/*
-jt-buffer/target/*
-jt-scheduler/target/*
-jt-binarize/target/*
-jt-bandcombine/target/*
-jt-convolve/target/*
-jt-classifier/target/*
-jt-piecewise/target/*
-jt-artifact/target/*
-jt-format/target/*
-jt-colorconvert/target/*
-jt-errordiffusion/target/*
-jt-orderdither/target/*
-jt-colorindexer/target/*
-jt-imagefunction/target/*
-jt-rlookup/target/*
-jt-vectorbin/target/*
-changematrix/target/*
-jt-concurrent-tile-cache/src/test/resources/it/geosolutions/logfiles
-jt-concurrent-tile-cache/target/*
-jt-concurrent-tile-cache/bin/*
-roiaware-legacy/target/*
-jt-squareroot/target/*
-release/target/*
-deploy/*
-/jt-shadedrelief/target/
\ No newline at end of file
+build-full.bat
+target
+.settings
+.classpath
+.project
+.metadata
+.DS_Store
+*.class
+*.pyc
+
+/.metadata/
+
+# Intellij
+.idea/
+*.iml
+
+# Cucumber Testing artifacts
+cucumber-report
+.mvn
+
+# NSG CITE testst data directory
+data/citensg-1.0/gwc/cite*
+data/citensg-1.0/gwc/geowebcache_*.bak
+data/citensg-1.0/legendsamples/
+data/citensg-1.0/logs/geoserver.log
```

Comment:
Yes, it is, otherwise the ignored files will be different on different platforms. 
GT and GS also have one, e.g.:
https://github.com/geotools/geotools/blob/master/.gitignore